pkg/ovn/routeadvertisement.go

```diff
@@ -0,0 +1,396 @@
+package ovn
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/golang/glog"
+	"github.com/rh-ecosystem-edge/eco-goinfra/pkg/clients"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/rh-ecosystem-edge/eco-goinfra/pkg/msg"
+	ovnv1 "github.com/rh-ecosystem-edge/eco-goinfra/pkg/schemes/ovn/routeadvertisement/v1"
+	"github.com/rh-ecosystem-edge/eco-goinfra/pkg/schemes/ovn/types"
+)
+
+// RouteAdvertisementBuilder provides a wrapper around RouteAdvertisements objects for the Kubernetes API.
+type RouteAdvertisementBuilder struct {
+	// RouteAdvertisement definition, used to create the RouteAdvertisement object.
+	Definition *ovnv1.RouteAdvertisements
+	// Created RouteAdvertisement object.
+	Object *ovnv1.RouteAdvertisements
+	// api client to interact with the kubernetes cluster.
+	apiClient client.Client
+	// Used to store latest error message upon defining or mutating RouteAdvertisement definition.
+	errorMsg string
+}
+
+// NewRouteAdvertisementBuilder creates a new instance of RouteAdvertisementBuilder.
+func NewRouteAdvertisementBuilder(
+	apiClient *clients.Settings,
+	name string) *RouteAdvertisementBuilder {
+	glog.V(100).Infof(
+		"Initializing new RouteAdvertisement structure with the following params: name: %s",
+		name)
+
+	if apiClient == nil {
+		glog.V(100).Infof("RouteAdvertisement 'apiClient' cannot be nil")
+
+		return nil
+	}
+
+	err := apiClient.AttachScheme(ovnv1.AddToScheme)
+	if err != nil {
+		glog.V(100).Infof("Failed to add ovn scheme to client schemes: %v", err)
+
+		return nil
+	}
+
+	builder := &RouteAdvertisementBuilder{
+		apiClient: apiClient.Client,
+		Definition: &ovnv1.RouteAdvertisements{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "RouteAdvertisements",
+				APIVersion: fmt.Sprintf("%s/%s", ovnv1.GroupName, ovnv1.SchemeGroupVersion.Version),
+			},
```

Comment:
Is this required? I don't think we set the type information manually in other packages